src/AutoRest.CSharp/Common/Input/Configuration.cs

```diff
@@ -174,7 +179,7 @@ public enum UnreferencedTypesHandlingOption
             _modelsToTreatEmptyStringAsNull = new HashSet<string>(modelsToTreatEmptyStringAsNull);
             _intrinsicTypesToTreatEmptyStringAsNull.UnionWith(additionalIntrinsicTypesToTreatEmptyStringAsNull);
             _methodsToKeepClientDefaultValue = methodsToKeepClientDefaultValue ?? Array.Empty<string>();
-            _apiTypes = branded ? new AzureApiTypes() : new SystemApiTypes();
+            _apiTypes =  "azure".Equals(flavor, StringComparison.InvariantCultureIgnoreCase) ? new AzureApiTypes() : new SystemApiTypes();
```

Comment:
I think it is case sensitive.

Reply:
> I think it is case sensitive.

It's not. Laurent has confirmed that, though I think it's better to be case sensitive.